### PR TITLE
[Merged by Bors] - feat: port Analysis.SpecialFunctions.ExpDeriv

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
This incorporates the changes from https://github.com/leanprover-community/mathlib/pull/19139 which haven't hit mathport yet due to mathport CI failing for a few days. (Hopefully it will succeed on the next run.)

---

- [x] depends on: #4642

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4642
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
@Parcly-Taxel, I got this working without having seen that you'd started the PR, so I've forced pushed.

**Comment by urkud** [comment]
I rebased on current `RealDeriv`. @semorrison I don't think that hijacking someone else's PR without asking is a good idea.

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean [hunk 1]
```diff
@@ -62,7 +61,8 @@ theorem iter_deriv_exp : ∀ n : ℕ, (deriv^[n]) exp = exp
 #align complex.iter_deriv_exp Complex.iter_deriv_exp
 
 theorem contDiff_exp : ∀ {n}, ContDiff 𝕜 n exp := by
-  refine' contDiff_all_iff_nat.2 fun n => _
+  apply contDiff_all_iff_nat.2
+  intro n
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean:65]: What was wrong with `refine`?

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean [hunk 2]
```diff
@@ -73,17 +73,13 @@ theorem contDiff_exp : ∀ {n}, ContDiff 𝕜 n exp := by
...
+  contDiff_exp.contDiffAt.hasStrictDerivAt' (hasDerivAt_exp x) le_rfl
 #align complex.has_strict_deriv_at_exp Complex.hasStrictDerivAt_exp
 
 theorem hasStrictFDerivAt_exp_real (x : ℂ) : HasStrictFDerivAt exp (exp x • (1 : ℂ →L[ℝ] ℂ)) x :=
   (hasStrictDerivAt_exp x).complexToReal_fderiv
 #align complex.has_strict_fderiv_at_exp_real Complex.hasStrictFDerivAt_exp_real
 
-theorem isOpenMap_exp : IsOpenMap exp :=
-  open_map_of_strict_deriv hasStrictDerivAt_exp exp_ne_zero
-#align complex.is_open_map_exp Complex.isOpenMap_exp
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean]: I see that you remove an import and a lemma. What's the plan here: move the lemma somewhere else (then you should open a `mathlib3` PR doing the same)? Restore it once the inverse function theorem is ported?

**Review by urkud — COMMENTED** [review]
Otherwise LGTM.

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4642~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean (hunk 2 above)
Comment (collares) [reply Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean]: I think that's forward porting leanprover-community/mathlib#19139

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean:65]: It wasn't type checking, something to do with implicit lambdas.

I'm out of action for the next 12 hours, so please anyone feel free to make fixes / porting notes.

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean]: I will rebase on top of a fresh `mathport` output in ~15min.

Comment (urkud) [reply Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean]: It didn't go through `mathport` yet. Should we manually update SHA sum in the header?

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean [hunk 3]
```diff
@@ -0,0 +1,323 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Abhimanyu Pallavi Sudhir, Jean Lo, Calle Sönne
+
+! This file was ported from Lean 3 source module analysis.special_functions.exp_deriv
+! leanprover-community/mathlib commit 2c1d8ca2812b64f88992a5294ea3dba144755cd1
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean:7]:
```suggestion
! leanprover-community/mathlib commit 6a5c85000ab93fe5dcfdf620676f614ba8e18c26
```

**Comment by urkud** [comment]
Please either apply the suggested SHA sum update or add a porting note/todo. Otherwise LGTM.
bors d+

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean]: It's gone through mathport now

File: Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Analysis/SpecialFunctions/ExpDeriv.lean]: I pushed a fix, it needed an `@`.

**Comment by eric-wieser** [comment]
I rebased on the latest mathport, and pushed one new commit.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**